packages/back-end/test/api/environments.test.ts

```diff
@@ -0,0 +1,775 @@
+import mongoose from "mongoose";
+import { MongoMemoryServer } from "mongodb-memory-server";
+import request from "supertest";
+import { getAuthConnection } from "../../src/services/auth";
+import authenticateApiRequestMiddleware from "../../src/middleware/authenticateApiRequestMiddleware";
+import app from "../../src/app";
+import { updateOrganization } from "../../src/models/OrganizationModel";
+import { findAllProjectsByOrganization } from "../../src/models/ProjectModel";
+
+jest.mock("../../src/models/ProjectModel", () => ({
+  findAllProjectsByOrganization: jest.fn(),
+}));
+
+jest.mock("../../src/models/OrganizationModel", () => ({
+  updateOrganization: jest.fn(),
+}));
+
+jest.mock("../../src/init/queue", () => ({
+  queueInit: () => undefined,
+}));
+
+jest.mock("../../src/services/auth", () => ({
+  ...jest.requireActual("../../src/services/auth"),
+  getAuthConnection: () => ({
+    middleware: jest.fn(),
+  }),
+}));
+
+jest.mock("../../src/middleware/authenticateApiRequestMiddleware", () => ({
+  ...jest.requireActual(
+    "../../src/middleware/authenticateApiRequestMiddleware"
+  ),
+  __esModule: true,
+  default: jest.fn(),
+}));
+
+let mongod;
+
+describe("environements API", () => {
+  const OLD_ENV = process.env;
+
+  beforeAll(async () => {
+    jest.resetModules();
+    mongod = await MongoMemoryServer.create();
+    const uri = mongod.getUri();
+    process.env.MONGO_URL = uri;
+    getAuthConnection().middleware.mockImplementation((req, res, next) => {
+      next();
+    });
+  });
+
+  afterAll(async () => {
+    await mongoose.connection.dropDatabase();
```

Comment:
What is this for?  Seems dangerous to have this here.  Feels like we should not reference `mongoose` directly here at all and instead do all of our operations on the `mongod` object, which we know is an in-memory mock server.

Reply:
Looks like you still have to also stop mongoose but the DB drop can be skipped of course.